src/main/java/com/service/OrderService.java

```diff
@@ -0,0 +1,126 @@
+package com.service;
+
+import com.dao.*;
+import com.model.*;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+public class OrderService {
+    private OrderDAO orderDAO = OrderDAO.getInstance();
+    private UserDAO userDAO = UserDAO.getInstance();
+    private DishDAO dishDAO = DishDAO.getInstance();
+    private DishOrderDAO dishOrderDAO = DishOrderDAO.getInstance();
+    private static OrderService instance;
+
+    private OrderService() {
+    }
+
+    public void setOrderDAO(OrderDAO orderDAO) {
+        this.orderDAO = orderDAO;
+    }
+
+    public void setUserDAO(UserDAO userDAO) {
+        this.userDAO = userDAO;
+    }
+
+    public void setDishDAO(DishDAO dishDAO) {
+        this.dishDAO = dishDAO;
+    }
+
+    public void setDishOrderDAO(DishOrderDAO dishOrderDAO) {
+        this.dishOrderDAO = dishOrderDAO;
+    }
+
+    public static OrderService getInstance() {
+        if (instance == null) {
+            instance = new OrderService();
+        }
+        return instance;
+    }
+
+    public boolean cancelOrder(Long id) {
+        return orderDAO.remove(id);
+    }
+
+    public void makeOrder(String userName, Map<String, Long> dishNamesAndAmount) {
+        if (!userDAO.getByName(userName).isPresent()) {
+            throw new NoSuchElementException("There is no such user!");
+        }
+        long userID = userDAO.getByName(userName).get().getId();
+        orderDAO.create(new Order(0, userID, LocalDateTime.now(), 0, Order.Status.CREATED));
```

Comment:
You should make return type of this method long (id that was generated for this order), so that you don't have to perform operations that you wrote later

Reply:
We thought about it at very beginning. In case of several simultaneous invocation of 'create' method calculated id will be wrong.
We chose the most obtainable way due to the deadline.